stl/inc/atomic

```diff
@@ -639,35 +633,36 @@ struct _Atomic_storage<_Ty, 8> { // lock-free using 8-byte intrinsics
         // non-atomically initialize this atomic
     }
 
-#ifdef _M_IX86
-    void store(const _Ty _Value, const memory_order _Order = memory_order_seq_cst) noexcept {
-        // store with (effectively) sequential consistency
-        _Check_store_memory_order(_Order);
-        (void) exchange(_Value, _Order);
-    }
-#else // ^^^ _M_IX86 / !_M_IX86 vvv
-
     void store(const _Ty _Value) noexcept { // store with sequential consistency
         const auto _Mem           = _Atomic_address_as<long long>(_Storage);
         const long long _As_bytes = _Atomic_reinterpret_as<long long>(_Value);
-#ifdef _M_ARM64
+#if defined(_M_IX86)
+        auto _Comparand = __iso_volatile_load64(_Mem);

```

Comment:
Why is this better than `_InterlockedExchange64`?

Reply:
Sorry, that's wrong branch, reverted

Reply:
But to answer your question, it is x86, where there's only `CMPXHG8B` for 64-bit types exchange.